cmake/common.cmake

```diff
@@ -0,0 +1,35 @@
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
+    
+string(ASCII 27 Esc)
+set(Green    "${Esc}[92m"     CACHE STRING "green color"    FORCE)
+set(Blue     "${Esc}[1;94m"   CACHE STRING "Blue color"     FORCE)
+set(BoldRed  "${Esc}[1;5;91m" CACHE STRING "bold-red color" FORCE)
+set(NoColor  "${Esc}[0m"      CACHE STRING "reset color"    FORCE)
+
+if(NOT WIN32)
+
+    execute_process(COMMAND   ${CMAKE_C_COMPILER}  -dumpversion   OUTPUT_VARIABLE   GCC_VERSION)
+    if(GCC_VERSION VERSION_GREATER 4.9 OR GCC_VERSION VERSION_EQUAL 4.9)
+        set(STDLIBC  "-std=c++1y")
```

Comment:
We use C++ so little in NLopt that the benefits of using features from the latest C++ standard don't seem to outweigh the cost of dropping support for usage by old programs that require earlier versions of the standard.